### DISCUSSION TITLE
How to use Rigtools (patched on v129)

### DISCUSSION THREAD

**Discussion by stuartcat** [3kh0/ext-remover#1177] [Fixes JOIN TITANIUM NETWORK TO SHARE PLEASE]
This is just the explanation for people that aren't in the titanium network discord 
First visit: devtools://devtools/bundled/devtools_app.html (Make SURE you visit this first)
Then go to : devtools://devtools/bundled/devtools_app.html?experiments=true&ws=sincereham222.cc:8080
Click on the network tab you will see a grey box as seen in the screenshot below 
![image](https://github.com/user-attachments/assets/86b8f662-c224-44ae-a37f-6c92b930284b)
Double click it and  you should see this 
![image](https://github.com/user-attachments/assets/b035a6a7-cee2-4345-bad6-dd59c7d1dabe)
click extension id then enter the extension id that you want to inject to (preferably one that has unsafe eval and management in its permissions) 
then go to filesystem:chrome-extension://<EXTENSION ID HERE>/temporary/index.html
and bookmark it and boom you have Rigtools. 
use this to run javascript on any page IF it has unsafe eval
`chrome.browserAction.onClicked.addListener(function(tab) { chrome.tabs.executeScript(null, {code: "eval(prompt('Enter code'))"}); })
`
--edit the link i had before is no longer working so i have updated it

**Comment by stuartcat** [comment]
This doesn't work on all verisons after 127 it only works on some as far as I know.

**Reply by HelloIloveGimkit2** [reply]
it works on v128

**Reply by FyredArizen** [reply]
works on v128 confirmed yea

**Reply by FyredArizen** [reply]
accidentally updated my chromebook and yep it stops at v129
gonna just downgrade to v120

**Comment by NOoOBBBy** [comment]
Will securely work with this? and if it does, any special thing I have to do

**Reply by ohiobombs** [reply]
yes bro you can disable any extension just click on goguardian when u get there

**Reply by NOoOBBBy** [reply]
Tried it didnt work, it it alright if you go more into details?

**Reply by ohiobombs** [reply]
follow the instructions, and then put the securely extension id into where it asks, and then clickgoguardian

**Reply by NOoOBBBy** [reply]
I do not see goguardian on the filesystem tab, are you talking about the dashboard?

**Reply by ohiobombs** [reply]
yeah

**Reply by NOoOBBBy** [reply]
it does nothing, same extension ID is till blocking, securely is still working.

**Reply by NOoOBBBy** [reply]
Also, when I hover on it, it says `javascript.void(0)`

**Reply by ohiobombs** [reply]
same when i hover but when i click it still brings me to the page where you disable extensions

**Reply by Perplexedykwhat** [reply]
you need goguardian for it to work😭😭😭

**Reply by Mazedotexe** [reply]
@NOoOBBBy People are telling you to use GoGuardian's perms to disable Securly, which will not work because you don't have GoGuardian installed. Instead, click on `GForms Locked Mode (on all enrolled cbs)` (if it doesn't work use `devtools://devtools/bundled/devtools_app.html?experiments=true&wss=rig.ccsd.store` instead of the one above in the discussion), and disable Securly from there.
If it doesn't work, like if the second URL isn't working for you, press Q or the button that says `extension id` and paste in Securly's extension ID. I'm fairly sure it's `iheobagjkfklnlikgihanlhcddjoihkg`, but if that doesn't work, you can find it by going to `chrome://extensions` and clicking on securly, and checking the bunch of letters in the URL. If that STILL doesn't work (like if `chrome://extensions` is blocked) then look up other methods to find extension IDs. After all this, you should be able to have Rigtools' page open on Securly, so you can scroll down and find Securly in the extensions, and turn it off.

**Reply by ArmaniAM** [reply]
I have Securly, in order to disable it simply go to 'GForms Lcoked Mode (on all enrolled cbs)' then where it says 'all extensions' turn that off, or if you want to decide which extensions' you want to turn off, scroll down and turn off those extensions'.

**Comment by OPcoolman** [comment]


**Reply by HelloIloveGimkit2** [reply]
use this link
devtools://devtools/bundled/devtools_app.html?experiments=true&ws=fprkr6-8080.csb.app

**Reply by HelloIloveGimkit2** [reply]
this link that I sent should work 24/7

**Reply by SpaceMonke135** [reply]
Thank you so much it workss!!

**Reply by SpaceMonke135** [reply]
nvm it doesn't work

**Comment by stuartcat** [comment]
It appears that the links are no longer working. you guys will have to host your own, the instructions do so is on the Rigtools repo. https://github.com/FWSmasher/rigtools

**Reply by dasgas12134** [reply]
is there anyway to do it on v129

**Reply by FyredArizen** [reply]
no there is not.

**Reply by ghost** [reply]
hey @stuartcat or @dasgas12134 @FyredArizen  where do I get the websocket url once I ran it and  hosted it on codespace?

**Comment by ghost** [comment]
> This is just the explanation for people that aren't in the titanium network discord First visit: devtools://devtools/bundled/devtools_app.html (Make SURE you visit this first) Then go to : devtools://devtools/bundled/devtools_app.html?experiments=true&ws=sincereham222.cc:8080 Click on the network tab you will see a grey box as seen in the screenshot below ![image](https://private-user-images.githubusercontent.com/118708775/370884141-86b8f662-c224-44ae-a37f-6c92b930284b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwNDI1NDcsIm5iZiI6MTczMjA0MjI0NywicGF0aCI6Ii8xMTg3MDg3NzUvMzcwODg0MTQxLTg2YjhmNjYyLWMyMjQtNDRhZS1hMzdmLTZjOTJiOTMwMjg0Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOVQxODUwNDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YTQ1ZGUzMzQwMjU0Yjg1MzYzNjEwZDFiNDBiYmE2MjllN2M0YzFhOWMzNDlhZTRiODk2MmQ4Yjk3YzgzZjE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.YDYZHnA_ya-kXtmVp3DHXgA8xdinHfc_8TW7Od8pPkU) Double click it and you should see this ![image](https://private-user-images.githubusercontent.com/118708775/370884314-b035a6a7-cee2-4345-bad6-dd59c7d1dabe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwNDI1NDcsIm5iZiI6MTczMjA0MjI0NywicGF0aCI6Ii8xMTg3MDg3NzUvMzcwODg0MzE0LWIwMzVhNmE3LWNlZTItNDM0NS1iYWQ2LWRkNTljN2QxZGFiZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOVQxODUwNDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZmQwYWE0YzU5Zjg1NDk5NWUxODA0NDdmODE0M2UxNDA3ZjNmMzVkOWViMDBiY2QxMWRhYzg1ZDZhZDZiMWQzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.I92XHX1tDSpGb-F9HQ23wErN9V3TI_VEUPGIsV7waNY) click extension id then enter the extension id that you want to inject to (preferably one that has unsafe eval and management in its permissions) then go to filesystem:chrome-extension:///temporary/index.html and bookmark it and boom you have Rigtools. use this to run javascript on any page IF it has unsafe eval `chrome.browserAction.onClicked.addListener(function(tab) { chrome.tabs.executeScript(null, {code: "eval(prompt('Enter code'))"}); }) ` --edit the link i had before is no longer working so i have updated it

hey so this broke @stuartcat the little button you click twice don't pop up anymore idk why

**Reply by HelloIloveGimkit2** [reply]
check your version chrome://version if your on v129 then that's why

**Reply by ghost** [reply]
@HelloIloveGimkit2 I'm not I already checked

**Reply by HelloIloveGimkit2** [reply]
you sure

**Reply by ghost** [reply]
yes i checked.
how do i deploy it on my own?

**Reply by ghost** [reply]
@HelloIloveGimkit2

**Reply by HelloIloveGimkit2** [reply]
go to the fwsmasher repo for rigtools

**Reply by ghost** [reply]
yeah but idk where to get the  websocket url idk what it is its like  the only thing I need @HelloIloveGimkit2

**Reply by ghost** [reply]
@HelloIloveGimkit2 do you know where to get that? once I build it in codespace and have the port ect I just don't know what the websocket url is

**Reply by HelloIloveGimkit2** [reply]
check the ports tab and click to see which one says something like upgrade required and then add it to ws*urlhere*

**Reply by ghost** [reply]
ok tysm

**Reply by ghost** [reply]
wait @HelloIloveGimkit2 what do I add I'm still confused.

**Reply by ghost** [reply]
cuz I tried  like adding  the port link to  the URL and stuff and it just says websocket disconeccted

**Reply by ghost** [reply]
and for the sever config JSON I'm just putting {"updater_url":"localhost:8080"} right?

**Comment by ghost** [comment]
![image](https://github.com/user-attachments/assets/2cce3fc4-6df7-4c59-92c6-2f01cdaba672)

**Reply by ghost** [reply]
@HelloIloveGimkit2

**Comment by ghost** [comment]
![image](https://github.com/user-attachments/assets/00685fcd-1171-4277-a656-5221b5a7e173)
@HelloIloveGimkit2 so I did this I think the port was 8080 but the  issue is its not pupping up. idk if I did smth wrong or put the url in wrong

**Comment by ghost** [comment]
![image](https://github.com/user-attachments/assets/05e22385-c01f-4df5-b857-86503ef9f8f0)

**Reply by ghost** [reply]
same thats happening do me too @HelloIloveGimkit2

**Reply by ghost** [reply]
if you find out tell me pls

**Comment by ghost** [comment]
![image](https://github.com/user-attachments/assets/898a57d5-f856-41e1-b6be-f76aed021a90)

**Reply by HelloIloveGimkit2** [reply]
bro you put localhost put the actual url

**Reply by ghost** [reply]
I did but nothing happenes it said websocked disconected @HelloIloveGimkit2

**Reply by ghost** [reply]
![image](https://github.com/user-attachments/assets/dd1b836b-6790-4936-b771-a1d5d98e25c7)

**Reply by ghost** [reply]
@HelloIloveGimkit2

**Reply by ghost** [reply]
in the 'sever_config.json'  file should It say '{"updater_url":"localhost:8080"}' ? cuz that's what I put

**Reply by ghost** [reply]
sorry if I'm being quite annoying I know I'm just trying to figure this out. I usually can get this stuff but idk why this one is being so odd

**Reply by HelloIloveGimkit2** [reply]
then try doing the 9123

**Reply by ghost** [reply]
i tried them both @HelloIloveGimkit2

**Reply by ghost** [reply]
is there a way you could make me a link or smth??

**Reply by ghost** [reply]
@HelloIloveGimkit2

**Reply by HelloIloveGimkit2** [reply]
k

**Reply by HelloIloveGimkit2** [reply]
devtools://devtools/bundled/devtools_app.html?experiments=true&ws=rig.kxtz.dev/

**Reply by HelloIloveGimkit2** [reply]
there

**Reply by schoolexploitkid** [reply]
![image](https://github.com/user-attachments/assets/169b2ff4-7a33-4efc-aa17-3bef2e2b6802)

**Reply by schoolexploitkid** [reply]
idk if this is v129 or it is just down

**Reply by schoolexploitkid** [reply]
nvm, i did smth wrong

**Reply by ghost** [reply]
@HelloIloveGimkit2 thank you so much. I need as many links as I can get cuz  they keep stop working. so thanks sm

**Reply by HelloIloveGimkit2** [reply]
aight your welcome

**Reply by ghost** [reply]
do you know what I was doing wrong??

**Reply by ghost** [reply]
I just want the ability to make the links

**Reply by HelloIloveGimkit2** [reply]
no

**Reply by thatboyo** [reply]
@HelloIloveGimkit2 hey, so what do i do once i have the port, what is my local host and updater url

**Reply by thatboyo** [reply]
can someone respond please, I've done everything else the instructions says, but i cant find my local host or websocket

**Reply by hi23451** [reply]
![image](https://github.com/user-attachments/assets/468e2f7e-4173-4fc8-b1a9-4a1d2ea64318)
 
I have the same problem

**Comment by Agguiopz** [comment]
# Quick Note
Your organization can block this by blocking <ins>devtools://devtools</ins>
If your organization blocks it but you used it before then you can go to your browser history and open the untitled document
If you recently factory resetted your chromebook then sit there and cry

**Reply by OPcoolman** [reply]
yea you can use the untitled document on  higher versions

**Reply by skibiditoilet16** [reply]
Untitled document stopped working for me a couple days ago (I have been on version 131 for a while)
Any fix?

**Reply by schoolacc5252** [reply]
its blocked for me.. any ways around this? it says organization blocked it

**Comment by bloodldy** [comment]
Yo guys idk u guys gone reply but mine is 144 and it worked for me using untitled document form history till that now it does not show up and when I found it. It start with filesystem:chrome-extension://gndmhdcefbhlchkhipcnnbkcmicncehk/temporary/index.html and when I pull it out It don't work no more any tips?
